sympy/solvers/solveset.py

```diff
@@ -539,23 +539,7 @@ def _solve_real_trig(f, symbol):
     """ Helper to solve trigonometric equations """
     f = trigsimp(f)
     f = f.rewrite(exp)
-    f = together(f)
-    g, h = fraction(f)
-    y = Dummy('y')
-    g, h = g.expand(), h.expand()
-    g, h = g.subs(exp(I*symbol), y), h.subs(exp(I*symbol), y)
-    if g.has(symbol) or h.has(symbol):
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
-
-    solns = solveset_complex(g, y) - solveset_complex(h, y)
-
-    if isinstance(solns, FiniteSet):
-        return Union(*[invert_complex(exp(I*symbol), s, symbol)[1]
-                       for s in solns])
-    elif solns is S.EmptySet:
-        return S.EmptySet
-    else:
-        return ConditionSet(symbol, Eq(f, 0), S.Reals)
```

Comment:
In [return statement](https://github.com/sympy/sympy/pull/10482/files#diff-eec0422923e8f100745c015cd8fdd6cfR542) we are passing `f` which is now in `exp` form and `domain= complexes` so `solveset` take this to [here](https://github.com/Shekharrajak/sympy/blob/master/sympy/solvers/solveset.py#L968) `solveset_complex`.I just wanted to make sure that if there is more complicated/nested `exp` form then it should able to handle that automatically.Isn't it ?